### PR TITLE
Update OpenTelemetry protos to v0.16.0

### PR DESCRIPTION
**Pull request by alanwest**
Three changes of note:

1) Replacing `InstrumentationLibrary` with `InstrumentationScope`. This should be a binary compatible change and not have any effect on OTLP consumers.
2) A number of deprecated fields have been removed - e.g., `IntSum`, `IntHistogram`, etc. We were not populating these anymore, so no impact on us.
3) The Name field on OTLP log records was removed. We were populating this with our `CategoryName`. We should decide what to do with this before merging this PR.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs [hunk 1]
```diff
@@ -41,15 +41,15 @@ internal static class LogRecordExtensions
             };
             request.ResourceLogs.Add(resourceLogs);
 
-            var instrumentationLibraryLogs = new OtlpLogs.InstrumentationLibraryLogs();
-            resourceLogs.InstrumentationLibraryLogs.Add(instrumentationLibraryLogs);
+            var scopeLogs = new OtlpLogs.ScopeLogs();
+            resourceLogs.ScopeLogs.Add(scopeLogs);
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs:45]: ILogger scope and Otel scopes 🤣

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs [hunk 2]
```diff
@@ -64,7 +64,6 @@ internal static OtlpLogs.LogRecord ToOtlpLog(this LogRecord logRecord)
                 otlpLogRecord = new OtlpLogs.LogRecord
                 {
                     TimeUnixNano = (ulong)logRecord.Timestamp.ToUnixTimeNanoseconds(),
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs:66]: lets just put a todo to store categoryName into attributes, to unblock this PR. OTLP LogExporter is non-stable anyway, so dont need to be blocked by this.

**Review by cijothomas — APPROVED** [review]
LGTM, with a todo for category name, and addition to changelog

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs (hunk 1 above)
Comment (alanwest) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs:45]: There is absolutely no way this will cause confusion. 🙄

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs (hunk 2 above)
Comment (reyang) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs:66]: +1

Comment (reyang) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs:66]: Maybe `Attributes["dotnet.ilogger.categoryname"]`, based on the spirit of https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#appendix-a-example-mappings and https://github.com/open-telemetry/opentelemetry-specification/blob/33e51e72276c4aa1dbe4fba84b4b05944ac5c241/specification/resource/semantic_conventions/README.md#telemetry-sdk.

Comment (cijothomas) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs:66]: or wait for resolution on https://github.com/open-telemetry/opentelemetry-specification/issues/2398

**Pull request merged**